### PR TITLE
event properties object instruction

### PR DESCRIPTION
**Pull request by yuhao900914**
# Amplitude Developer Docs PR

## Description

Describe your changes. 
Add instructions for the GTM event properties object input and identify bulk set operation.
Screenshot 2024-04-26 at 10 27 49 AM
Screenshot 2024-04-26 at 10 40 47 AM

## Deadline

When do these changes need to be live on the site?

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/sources/google-tag-manager-client.md [hunk 1]
```diff
@@ -196,13 +196,17 @@ The `track` tag type is for tracking an event under a specific trigger.
...
+##### Event Properties Basic
 
 | Name  | Description |
 | --- | --- |
 | `Property Name` | `string`. The name of the event property. |
 | `Property Value` | `string`. The value of the event property. |
 
+##### Event Properties Object
+
+Choose the object format for the GTM variable to define event properties. It's important to note that if there are any duplicate keys, the event properties object will overwrite the basic event properties. Any input not in object format will be ignored.
```
Comment (markzegarelli) [inline docs/data/sources/google-tag-manager-client.md:208]:
```suggestion
Choose the object format for the GTM variable to define event properties. Use unique values for keys in the object. Amplitude overwrites any basic event properties that have duplicate keys. Amplitude ignores any inputs not in the object format.
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]

File: docs/data/sources/google-tag-manager-client.md (hunk 1 above)
Comment (yuhao900914) [reply docs/data/sources/google-tag-manager-client.md:208]: Thanks for reviewing @markzegarelli .
Basically, there are 2 event properties. **EVENT PROPERTIES BASIC** and **EVENT PROPERTIES OBJECT**. I What I want to clarify here is if there is a duplicate key between **EVENT PROPERTIES BASIC** and **EVENT PROPERTIES OBJECT**, the value in **EVENT PROPERTIES OBJECT** will overwrite the value in **EVENT PROPERTIES BASIC**.  

I feel here is a misunderstanding.

File: docs/data/sources/google-tag-manager-client.md [hunk 2]
```diff
@@ -196,13 +196,24 @@ The `track` tag type is for tracking an event under a specific trigger.
...
+##### Event Properties Basic
 
 | Name  | Description |
 | --- | --- |
 | `Property Name` | `string`. The name of the event property. |
 | `Property Value` | `string`. The value of the event property. |
 
+##### Event Properties Object
+
+Select a GTM variable that returns a valid event properties object. This will overwrite the **Event Properties Basic** event properties if there has any duplicate key. Amplitude ignores any inputs not in the object format.
```
Comment (markzegarelli) [inline docs/data/sources/google-tag-manager-client.md:208]:
```suggestion
Select a GTM variable that returns a valid event properties object. This overwrites the **Event Properties Basic** event properties if there are any duplicate keys. Amplitude ignores any inputs not in the object format.
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]

**Review by markzegarelli — APPROVED** [review]

File: docs/data/sources/google-tag-manager-client.md (hunk 1 above)
Comment (yuhao900914) [reply docs/data/sources/google-tag-manager-client.md]: Thanks @markzegarelli.
The feature has not been published yet, I will merge this PR after the feature release.

File: docs/data/sources/google-tag-manager-client.md [hunk 3]
```diff
@@ -196,13 +196,24 @@ The `track` tag type is for tracking an event under a specific trigger.
 | --- | --- |
 | `Event Type` | `string`. The name of the event. |
 
-##### Event Properties
+##### Event Properties Basic
```
Comment (izaaz) [inline docs/data/sources/google-tag-manager-client.md:199]: Should we call this Event properties key value pairs?

File: docs/data/sources/google-tag-manager-client.md [hunk 4]
```diff
@@ -196,13 +196,24 @@ The `track` tag type is for tracking an event under a specific trigger.
...
 
 | Name  | Description |
 | --- | --- |
 | `Property Name` | `string`. The name of the event property. |
 | `Property Value` | `string`. The value of the event property. |
 
+##### Event Properties Object
+
+Select a GTM variable that returns a valid event properties object. This overwrites the **Event Properties Basic** event properties if there are any duplicate keys. Amplitude ignores any inputs not in the object format and any value under `user_properties` key.
```
Comment (izaaz) [inline docs/data/sources/google-tag-manager-client.md:208]: Can we call out the Google variable that has both event properties and user properties in it? And provide an example of what that looks like and what will be ignored in that object.

Comment (izaaz) [reply docs/data/sources/google-tag-manager-client.md:208]: And actually, we should just advice that they use that variable always to keep it simple.

File: docs/data/sources/google-tag-manager-client.md (hunk 3 above)
Comment (yuhao900914) [reply docs/data/sources/google-tag-manager-client.md:199]: I think both ways are setting the key-value pair. But the difference is it's by individual or bulk set. 
I use individual operation vs Bulk set operation for the identify properties. How do you like it?
May be here we use  **Individual Event Properties**?

File: docs/data/sources/google-tag-manager-client.md (hunk 4 above)
Comment (yuhao900914) [reply docs/data/sources/google-tag-manager-client.md:208]: Added another shared section to introduce **Google Tag: Event Settings** and added it to both the user properties and event properties section.

File: docs/data/sources/google-tag-manager-client.md (hunk 3 above)
Comment (izaaz) [reply docs/data/sources/google-tag-manager-client.md:199]: Yea maybe lets go with individual event properties.

File: docs/data/sources/google-tag-manager-client.md [hunk 5]
```diff
@@ -196,13 +196,28 @@ The `track` tag type is for tracking an event under a specific trigger.
...
+##### Individual Event Properties
 
 | Name  | Description |
 | --- | --- |
 | `Property Name` | `string`. The name of the event property. |
 | `Property Value` | `string`. The value of the event property. |
 
+##### Event Properties Object
+
+Select a GTM variable that returns a valid event properties object. This overwrites the **Event Properties Basic** event properties if there are any duplicate keys. Amplitude ignores any inputs not in the object format and any value under `user_properties` key.
```
Comment (izaaz) [inline docs/data/sources/google-tag-manager-client.md:208]:
```suggestion
Select a GTM variable that returns a valid event properties object. This overwrites the **Individual event properties** if there are any duplicate keys. Amplitude ignores any inputs not in the object format and any value under `user_properties` key.
```

**Review by izaaz — APPROVED** [review]

File: docs/data/sources/google-tag-manager-client.md (hunk 5 above)
Comment (yuhao900914) [reply docs/data/sources/google-tag-manager-client.md]: fixed.

**Pull request merged**